jxmlease/_basenode.py

```diff
@@ -989,35 +992,73 @@ def emit_handler(self, content_handler, pretty=True, newl='\n',
+                # document.
+                if len(curnode) > 1:
+                    full_document_ok = False
+                else:
+                    if isinstance(curnode, XMLListNode):
+                        curnode = curnode[0]
+                    elif isinstance(curnode, XMLDictNode):
+                        curnode = curnode[[i for i in curnode][0]]
+            elif isinstance(curnode.tag, (_unicode, str)):
+                # We have a tag. That will produce a top-level tag.
+                full_document_ok = True
+            else:
+                # We generally shouldn't get here, but let the user
+                # do what they want.
+                full_document_ok = True
+
+        if full_document is not None:
+            # We were given a full_document argument. Make sure it is
+            # OK to treat this as a "full document".
+            if full_document and not full_document_ok:
                 raise ValueError("Document will have more than one root node. "
                                  "The full_document argument must be False.")
-        elif isinstance(self, XMLListNode) and len(self) > 1:
-            # We have multiple tags. We cannot be a full document.
-            full_document = False
-        elif (self.tag is None and
-              isinstance(self, (XMLDictNode, XMLListNode)) and
-              len(self) <= 1):
-            # We are the root. We have a single root tag.
-            full_document = True
-        elif (isinstance(self.tag, (_unicode, str)) and
-              (self.parent is None or
-               (self.parent.tag is None and len(self.parent) == 1))):
-            # We look like the only child of the root node. In other words,
-            # we look like the root tag. Treat us as a full document.
-            full_document = True
         else:
-            # We don't appear to be the root, or it appears there is a multi-
-            # member root.
-            full_document = False
+            # Default to full_document=True, if it is OK to treat this
+            # as a full document and it looks like we are at the root.
+            full_document = full_document_ok
+            if full_document:
+                # Guess that we are at the root if:
+                # - self.tag is None, or
+                # - self.tag is not None, but we have no parent, or
+                # - self.tag is not None, but our parent looks like a
+                #   single-member, tagless root
+                if (self.tag is not None and self.parent is not None and
+                        (self.parent.tag is not None or len(self.parent) > 1)):
+                    full_document = False
 
         if full_document:
             content_handler.startDocument()
-        self._emit_handler(content_handler, depth=0, pretty=pretty, newl=newl,
-                           indent=indent)
+        curnode._emit_handler(content_handler, depth=0, pretty=pretty, newl=newl,
```

Comment:
You are enforcing a style standard to which we have not agreed. ;-)

FWIW, I believe pylint is set to 100 characters. If you want to enforce 80, we should do that globally.